src/features/courses/course-route.ts

```diff
@@ -371,14 +370,19 @@ router.put('/question/grade/:id',
     // This is to work around "extractMap" error
     // eslint-disable-next-line @typescript-eslint/no-explicit-any
     asyncHandler(async (req: RederlyExpressRequest<any, unknown, UpdateGradeRequest.body, UpdateGradeRequest.query>, _res: Response, next: NextFunction) => {
+        if (_.isNil(req.session)) {
```

Comment:
Should we move this up to a middleware? It needs to be everywhere except login/public routes.

Reply:
We cannot, I think there is a comment on the constant but this error can never be triggered, middleware is actually the problem here

The authentication middleware's existence enforces that req.session is not nil, however use strict does not know about checks in other closures

You'll see this elsewhere in the application where we have a closure and we have to have a nil check that can never actually be called